Projects/TestProject/RelationsTests.cs

```diff
@@ -0,0 +1,299 @@
+    }
+
+    [TestMethod]
+    public void TestRelationsPageSizeOne()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelationsPageSize( 1 );
+      qb.SetRelated( new List<String> { "Related" } );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "Order" ).Find( qb );
+      Object[] rel = (Object[]) res[ 0 ][ "Related" ];
+
+      Assert.IsTrue( rel.Length == 1 );
+    }
+
+    [TestMethod]
+    public void TestRelationPageSizeZero()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelationsPageSize( 0 );
+      qb.SetRelated( new List<String> { "Related" } );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "Order" ).Find( qb );
+      Object[] rel = (Object[]) res[ 0 ][ "Related" ];
+
+      Assert.IsTrue( rel.Length == 0 );
+    }
+
+    [TestMethod]
+    public void TestRelationsWithClass()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelated( new List<String> { "Related" } );
+      IList<Order> res = Backendless.Data.Of<Order>().Find( qb );
+
+      Assert.IsTrue( res[0].Related.Count == 1 );
+    }
+
+    [TestMethod]
+    public void TestSort()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelated( new List<String> { "Related" } );
+      qb.SetSortBy( new List<String> { "age" } );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "Order" ).Find( qb );
+
+      Assert.IsTrue( (Double) res[ 0 ][ "age" ] < (Double) res[ 1 ][ "age" ] );
+    }
+
+    [TestMethod]
+    public void TestWhereClauseZero()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetWhereClause( "name='Joe'" );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "Order" ).Find( qb );
+      Assert.IsTrue( res.Count == 0 );
+    }
+
+    [TestMethod]
+    public void TestWhereClauseNotZero()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetWhereClause( "Percentage > 30" );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "CountryLanguage" ).Find( qb );
+
+      foreach( Dictionary<String, Object> entry in res )
+        Assert.IsTrue( (Double) entry[ "Percentage" ] > 30.0 );
+    }
+
+    [TestMethod]
+    public void TestRelationsDepth()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelationsDepth( 1 );
+      qb.SetRelated( new List<String> { "Country", "Capital" } );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "CountryLanguage" ).Find( qb );
+
+      if(res[ 0 ].ContainsKey( "Country" ) )
+      {
+        Object entry = res[ 0 ][ "Country" ];
+        Assert.IsFalse( ( (Dictionary<Object, Object>) entry ).ContainsKey( "Capital" ) );
```

Comment:
What are you testing here? If object has Country and doesn't have Capital, the test passes? Why are you requesting `{ "Country", "Capital" }` ? Is "Capital" a relation column in "CountryLanguage" ?

Reply:
If object has Country and doesn't have Capital, the test passes? - Yes.

Why are you requesting { "Country", "Capital" } ? - This is error, i fixed it.

Is "Capital" a relation column in "CountryLanguage" ? - No, "Capiral", - a relation column in "Country".

What are you testing here? - I testing "relationsDepth field". When I changed the data type, I could break something, this test checks if the expected value of the "RelationsDepth" matches the real